### PR TITLE
Api - Request refactored

### PR DESCRIPTION
**Pull request by HansDahle**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**
Updates the implementation flow of requests. Instead of the logic being triggered when the request is created, it is now manually triggered by a start action.

**Testing:**
- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

TBD

**Checklist:**
- [x] Considered automated tests
- [ ] Considered updating specification / documentation
- [ ] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Domain/Queries/GetResourceOwner.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        public Guid AzureUniqueId { get; set; }
+
+        public class Handler : IRequestHandler<GetResourceOwner, FusionPersonProfile?>
+        {
+            private readonly HttpClient client;
+            private readonly IFusionProfileResolver profileResolver;
+
+            public Handler(IHttpClientFactory httpClientFactory, IFusionProfileResolver profileResolver)
+            {
+                this.client = httpClientFactory.CreateClient("lineorg");
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Domain/Queries/GetResourceOwner.cs:28]: clientname - constant or not ?

File: src/backend/api/Fusion.Resources.Logic/Errors/InvalidWorkflowError.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Fusion.Resources.Logic
+{
+    public class InvalidWorkflowError : ValidationException
+    {
+        //public List<WorkflowFieldError> FieldErrors { get; } = new List<WorkflowFieldError>();
+
+        public string WorkflowTypeName { get; set; }
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Logic/Errors/InvalidWorkflowError.cs:14]: Nullable ?

**Review by xssic — APPROVED** [review]
Some minor suggestions/questions, but nothing critical 👍

File: src/backend/api/Fusion.Resources.Api/Controllers/Requests/Requests/PatchInternalRequestRequest.cs [hunk 3]
```diff
@@ -11,7 +11,7 @@ namespace Fusion.Resources.Api.Controllers
 {
     public class PatchInternalRequestRequest : PatchRequest
     {
-        public PatchProperty<bool> IsDraft { get; set; } = new();
+        //public PatchProperty<bool> IsDraft { get; set; } = new();
```
[the remark this answers is not in the document]
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Controllers/Requests/Requests/PatchInternalRequestRequest.cs:14]: yep

File: src/backend/api/Fusion.Resources.Domain/Commands/Requests/InternalRequests/UpdateInternalRequest.cs [hunk 4]
```diff
@@ -50,13 +50,19 @@ public async Task<QueryResourceAllocationRequest> Handle(UpdateInternalRequest r
                 modified |= request.AssignedDepartment.IfSet(dep => dbRequest.AssignedDepartment = dep);
                 modified |= request.AdditionalNote.IfSet(note => dbRequest.AdditionalNote = note);
                 modified |= request.ProposedChanges.IfSet(changes => dbRequest.ProposedChanges = changes.SerializeToString());
-                modified |= request.IsDraft.IfSet(d => dbRequest.IsDraft = d);
+                //modified |= request.IsDraft.IfSet(d => dbRequest.IsDraft = d);
```
[the remark this answers is not in the document]
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Domain/Commands/Requests/InternalRequests/UpdateInternalRequest.cs:53]: yeo

File: src/backend/api/Fusion.Resources.Domain/Queries/GetResourceOwner.cs (hunk 1 above)
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Domain/Queries/GetResourceOwner.cs:28]: this is a bit hacky.. should create integration lib or include in people api

File: src/backend/api/Fusion.Resources.Logic/Errors/InvalidWorkflowError.cs (hunk 2 above)
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Logic/Errors/InvalidWorkflowError.cs:14]: Did that in the change request branch :)

**Pull request merged**